### PR TITLE
fix: support array types for most resources

### PR DESCRIPTION
**Pull request by nejch**
Closes https://github.com/python-gitlab/python-gitlab/issues/780.

I skipped `array of hashes` and I didn't touch some endpoints like merge requests that I think need another look at this topic (merge requests in particular). It's already a chunky PR, so I split it into fix/test/docs commits.

### CODE REVIEW

File: docs/cli-examples.rst [hunk 1]
```diff
@@ -119,14 +125,55 @@ Create a deploy token for a project:
...
+
+.. code-block:: console
+
+   $ gitlab -v project-access-token list --project-id 3
+
+Revoke a project access token:
+
+.. code-block:: console
+
+   $ gitlab project-access-token list --project-id 3 --id 1
```
Comment (JohnVillalovos) [inline docs/cli-examples.rst:156]: This doesn't seem to be a revoke

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]
Wow! That looks like it took a lot of time to go through all the docs!! Thanks.

Left comments on two things I saw.

File: docs/cli-examples.rst (hunk 1 above)
Comment (nejch) [reply docs/cli-examples.rst]: Oops thanks, copy paste again! :)

File: docs/cli-examples.rst [hunk 2]
```diff
@@ -119,14 +125,55 @@ Create a deploy token for a project:
...
+
+.. code-block:: console
+
+   $ gitlab -v group-access-token list --group-id 3
+
+Revoke a group access token:
+
+.. code-block:: console
+
+   $ gitlab group-access-token list --group-id 3 --id 1
```
[the remark this answers is not in the document]
Comment (nejch) [reply docs/cli-examples.rst]: Done as well

**Review by JohnVillalovos — APPROVED** [review]
Thanks @nejch

**Pull request merged**